### DISCUSSION TITLE
A '`HOMEBREW_UPDATE_REPORT_ALL`' Flag After Homebrew/brew#12910?

### DISCUSSION THREAD

**Discussion by RandomDSdevel** [Homebrew/discussions#3008] [Everyday usage] [answered]
(_Poster's note:_  Placed here to keep discussion separate from the referenced thread, at least to start with.)  

@MikeMcQuaid:  

&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Regarding Homebrew/brew#12910, could we potentially have a '`HOMEBREW_UPDATE_REPORT_ALL`' flag or similar to allow those who so wish to keep exercising the old behavior or, more likely, an equivalent?  Some folks, me included, might like seeing every formula that's been touched by a '`brew update`' run.  (Any other efficiency gains in this area and others, if any, are, naturally, still very much welcome, of course.  From what little I understand of the code there, though, that might be the best it can do, at least not without further refactoring…?  The list(s) of modified formulas and casks is/are already being built on the fly.)

**Comment by MikeMcQuaid** [comment]
I'd consider adding this if I see lots of people (10+) requesting this behaviour.

**Reply by SkittishSloth** [reply]
+1 request here.

Alternatively, if it isn't considered appropriate for brew to do this itself, is there some other means of generating a similar report? I'm not familiar enough with the system - is it just reading a git log or something?

**Reply by MikeMcQuaid** [reply]
@SkittishSloth Yeh, it's a fairly simple `git` operation: when `brew update` says the old and new revisions if you typed `git -C "$(brew --repo homebrew/core)" diff --stat abcd1234` (where `abcd1234` is the old revision that `brew update` output for Homebrew/homebrew-core) you'll see essentially the same information.

**Reply by tympom** [reply]
+1 for the old behaviour

**Reply by MikeMcQuaid** [reply]
@SkittishSloth @tympom Can you explain _why_ you like the old behaviour and what you use it for? It helps inform the decision here, thanks!

**Reply by tympom** [reply]
Yes, I like it as a discovery feature - what exists, what gets updated, added etc.
A choice woud be ideal, the short defaut is fine for me if there is an option to restore the full update view for those who are interested.

**Comment by MikeMcQuaid** [comment, marked as answer]
This change was reverted. Thanks for the feedback everyone!

**Reply by tympom** [reply]
Was this change introduced again lately?

**Reply by MikeMcQuaid** [reply]
Yes, on `master` but it's not in a tagged release yet. You can now opt-out of this behaviour with `HOMEBREW_UPDATE_REPORT_ALL_FORMULAE`. You may also want to investigate `HOMEBREW_UPDATE_REPORT_VERSION_CHANGED_FORMULAE` which has also become opt-in behaviour.

**Reply by tympom** [reply]
Great, thank you for the info 😀
